### PR TITLE
Dark Mode: Mark IV

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
- Our UIColor+Simplenote extension now relies on AssetCatalog to resolve dynamic colors
- We're replicating the `UIColorName` + `UIColor+Simplenote` mechanisms for UIImage(s).
- Lots of colors have been remapped to generic ones. **WHY:** Because the actual payload was duplicated!
- New Unit Tests!

Ref. #367

### Re-Mapped Colors:
```
.actionSheetFontColor = .textColor
.actionSheetBackgroundColor = .backgroundColor
.actionViewToggleTintColor = .lightBlueColor
.barTintColor	= .backgroundCollor
.collaboratorCellPrimaryLabelFontColor = .textColor
.collaboratorTextFieldTextColor = .textColor
.noteCellBackgroundSelectionColor = .lightBlueColor
.searchBarFontColor = .textcolor
.tagListSeparatorColor	= .dividerColor
.tagListFontColor = .textColor
.tagListHighlightBackgroundColor = .lightBlueColor
.tagListFontHighlightColor = .tintColor
.tagViewDeletionBackgroundColor = .lightBlueColor
.tableViewSeparatorColor = .dividerCcolor
.tableViewTextLabelColor = .textColor
.tableViewCellBackgroundHighlightColor = .lightBlueColor
.versionPickerDateFontColor = .textColor
```

### Nuked DB5 Entries:
```
.tableViewCellBackgroundColor
.barButtonFontColor
.barButtonSpacing
.barButtonSwapDuration
.editorStatusBarAlpha
.cellPanning*
.actionSheetTextFieldBackgroundColor
```

### Testing:
We need to verify Simplenote reacts correctly to **Light <> Dark Mode** switches.

- [x] Verify the Pinned Icon (Note List) is properly refreshed
- [x] Verify the Action Sheet looks great (Note Details > Top Right (i) iccon)
- [x] Verify the Horizontal Versions slider is properly refreshed
- [x] Verify the Notes List left divider is also refreshed
- [x] Verify that the Markdown Preview UI adapts automatically to any Style changes

### CODE REVIEW

**Comment by jleandroperez** [comment]
@bummytime Issues 1-3 fixed!!

Issue 4 to be addressed in PR Mark 5. This one will be merged back into develop **ONLY** after the last piece is fully worked out (the new UI switch!).

Ready for another look. Thank you!!

File: config/SDKVersions.xcconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// NOTE:
+// When something can be complex, it usually turns out being insanely complex.
```
Comment (bummytime) [inline config/SDKVersions.xcconfig]: 🤣

File: config/SDKVersions.xcconfig [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// We need two different approaches to detect Xcode11:
```
Comment (bummytime) [inline config/SDKVersions.xcconfig]: 😢

**Comment by jleandroperez** [comment]
@bummytime **Thank you** for reviewing this one!!!

**Pull request merged**